providers/google/tests/unit/google/cloud/hooks/test_bigquery.py

```diff
@@ -673,11 +676,25 @@ def test_job_id_validity(self, mock_md5, test_dag_id, expected_job_id):
             job_id=None,
             dag_id=test_dag_id,
             task_id="test_job_id",
-            logical_date=datetime(2020, 1, 23),
+            logical_date=None,
             configuration=configuration,
+            run_after=datetime(2020, 1, 23),
         )
         assert job_id == expected_job_id
 
+    def test_get_run_after_or_logical_date(self):
+        if AIRFLOW_V_3_0_PLUS:
+            from airflow.models import DagRun
+            from airflow.sdk.definitions.context import Context
+
+            ctx = Context(dag_run=DagRun(run_after=pendulum.datetime(2025, 1, 1)))
+            assert self.hook.get_run_after_or_logical_date(ctx) == pendulum.datetime(2025, 1, 1)
+        else:
+            from airflow.utils.context import Context
+
+            ctx = Context(logical_date=pendulum.datetime(2025, 1, 1))
+            assert self.hook.get_run_after_or_logical_date(ctx) == pendulum.datetime(2025, 1, 1)
```

Comment:
```suggestion
        if AIRFLOW_V_3_0_PLUS:
            from airflow.models import DagRun

            ctx = Context(dag_run=DagRun(run_after=pendulum.datetime(2025, 1, 1)))
        else:
            ctx = Context(logical_date=pendulum.datetime(2025, 1, 1))

        assert self.hook.get_run_after_or_logical_date(ctx) == pendulum.datetime(2025, 1, 1)
```

let's move the context import to top like `providers/google/tests/unit/google/cloud/operators/test_workflows.py`